packages-exp/auth-exp/test/helpers/integration/emulator_rest_helpers.ts

```diff
@@ -29,19 +41,47 @@ interface VerificationCodesResponse {
   verificationCodes: VerificationSession[];
 }
 
+export interface OobCodeSession {
+  email: string;
+  requestType: string;
+  oobCode: string;
+  oobLink: string;
+}
+
+interface OobCodesResponse {
+  oobCodes: OobCodeSession[];
+}
+
 export async function getPhoneVerificationCodes(
   auth: Auth
 ): Promise<Record<string, VerificationSession>> {
   assertEmulator(auth);
   const url = getEmulatorUrl(auth, 'verificationCodes');
-  const response: VerificationCodesResponse = await (await fetch(url)).json();
+  const response: VerificationCodesResponse = await (
+    await FetchProvider.fetch()(url)
+  ).json();
 
   return response.verificationCodes.reduce((accum, session) => {
     accum[session.sessionInfo] = session;
     return accum;
   }, {} as Record<string, VerificationSession>);
 }
 
+export async function getOobCodes(
+  auth: Auth
+): Promise<Record<string, OobCodeSession>> {
+  assertEmulator(auth);
+  const url = getEmulatorUrl(auth, 'oobCodes');
+  const response: OobCodesResponse = await (
+    await FetchProvider.fetch()(url)
+  ).json();
+
+  return response.oobCodes.reduce((accum, session) => {
+    accum[session.email] = session;
```

Comment:
This will be flaky if more than one email is sent to the same address (e.g. the recover OOB when changing emails). I'd advise keeping an array and asserting the array length.

Reply:
Done.